src/charm.py

```diff
@@ -193,6 +192,7 @@ def __init__(self, *args):
         self.framework.observe(self.on.start, self._on_start)
         self.framework.observe(self.on.stop, self._on_stop)
         self.framework.observe(self.on.remove, self._on_remove)
+        self.framework.observe(self.cert.on.cert_changed, self._on_cert_changed)
```

Comment:
If this is like the cos charm, the parent object already has an `_on_cert_changed` method and already calls it on `cert_changed`.
So this is effectively going to call it twice.

The right approach then is to remove this `observe()` call, and instead override `_on_cert_changed` with:

```
    def _on_cert_changed(self, event):
        """Event handler for cert change."""
        super()._on_cert_changed(event)
        self.update_tracing_receivers()
```